File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

const RoomChannel = consumer.subscriptions.create("RoomChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    let commentArea = document.getElementById('comments');

    commentArea.innerHTML += data['render'];
    console.log(data);
  },

  speak: function(comment, roomId) {
    return this.perform('speak', {
      comment: comment,
      room_id: roomId
    });
  }
});


let commentArea = document.getElementById('comment_area');

commentArea.addEventListener('keypress', (event) => {
  if (event.keyCode === 13) {

    let roomId = document.getElementById('room_id').value;

    RoomChannel.speak(event.target.value, roomId);
    event.target.value = '';
    return event.preventDefault();
  }
});

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel
  def subscribed
    stream_from "room_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    Comment.create!(content: data['message'], room_id: data['room_id'])
    ActionCable.server.broadcast 'room_channel', message: data['message']
  end
end
